packages/flutter_goldens/lib/src/flutter_goldens_io.dart

```diff
@@ -20,28 +22,110 @@ export 'package:flutter_goldens_client/skia_client.dart';
 
 const String _kFlutterRootKey = 'FLUTTER_ROOT';
 
+/// Similar to [matchesGoldenFile] but specialized for Flutter's own tests.
+///
+/// For descriptions of [key] and [version] parameters see [matchesGoldenFile].
+///
+/// If [isFlaky] is false, the golden check fails if the screenshot generated
+/// by the test does not match the golden stored in Skia Gold.
+///
+/// If [isFlaky] is true, the golden check will always pass no matter how many
+/// pixels deviate from the golden, and no matter how big the color difference
+/// is per pixel.
+///
+/// Whether [isFlaky] is true or false, the generated golden is always uploaded
+/// to Skia Gold for manual inspection. Use [isFlaky] to skip a flaky golden,
+/// and monitor it using the Skia Gold UI, by visiting https://flutter-gold.skia.org/list,
+/// and clicking on the respective golden test name. The UI will show the
+/// history of generated goldens over time. Each unique golden gets a unique
+/// color. If the color is the same for all commits in the recent history, the
+/// golden is likely no longer flaky and [isFlaky] can be set back to false. If
+/// the color changes from commit to commit then it is still flaky.
+AsyncMatcher matchesFlutterGolden(Object key, { int? version, bool isFlaky = false }) {
```

Comment:
Also, the flutter/flutter wiki page on golden file testing for the framework will need to be updated.

Reply:
Cool. I'm going to add it to my TODO list, but will wait until we converge on the overall design before starting the migration. Also, I imagine migrating all tests should probably done incrementally, since there are many usages of `matchesGoldenFile`.